backend/models/sequenceActivities.js

```diff
@@ -0,0 +1,298 @@
+}).noUnknown();
+
+const activityTreeSchema = yup.array().of(activitySchema).required();
+
+const activityCol = (modelId) => `${modelId}.activities`;
+
+const cleanActivityDetail = (activity) => {
+	activity._id = utils.uuidToString(activity._id);
+	activity.sequenceId = utils.uuidToString(activity.sequenceId);
+
+	if (activity.parent) {
+		activity.parent = utils.uuidToString(activity.parent);
+	}
+
+	return activity;
+};
+
+const fillIds = (activityTrees, sequenceId) => {
+	activityTrees.forEach((activity) => {
+		const _id = utils.stringToUUID(nodeuuid());
+
+		activity._id = _id;
+
+		if (activity.subActivities) {
+			fillIds(activity.subActivities, sequenceId);
+		}
+	});
+};
+
+const unrollActivities = (activityTrees, sequenceId, parentId, plainActivities = []) => {
+	activityTrees.forEach((activity) => {
+		const plainActivity = pick(activity,"_id", "name", "startDate", "endDate", "data", "resources");
+		plainActivity.sequenceId = sequenceId;
+
+		if (parentId) {
+			plainActivity.parent = parentId;
+		}
+
+		plainActivities.push(plainActivity);
+
+		if (activity.subActivities) {
+			unrollActivities(activity.subActivities, sequenceId, activity._id, plainActivities);
+		}
+
+	});
+
+	return plainActivities;
+};
+
+const simplifyActivity = (activity, subActivities) => {
+	const id = activity._id ? utils.uuidToString(activity._id) : null;
+
+	if (subActivities && subActivities.length) {
+		subActivities = { subActivities };
+	} else {
+		subActivities = {};
+	}
+
+	return  { id, ...pick(activity, "name", "startDate", "endDate"), ...subActivities };
+};
+
+const simplifyActivityTree = (activities) => {
+	return activities.map(activity => {
+		const subActivities = activity.subActivities ? simplifyActivityTree(activity.subActivities) : null;
+		return simplifyActivity(activity, subActivities);
+	});
+};
+
+const getDescendantsIds = (activities, parentId, ids = []) => {
+	activities.forEach(activity => {
+		if (utils.uuidToString(activity.parent) === parentId) {
+			ids.push(activity._id);
+			getDescendantsIds(activities, utils.uuidToString(activity._id), ids);
+		}
+	});
+
+	return ids;
+};
+
+/**
+ * @typedef {{_id: string, parents: Array<string>}} Activity
+ * @param {string} parentId
+ * @param {Array<Activity>} activities;
+ * @return {Array<Activity>}
+ */
+const getSubActivities = (parentId, activities) => {
+	return activities.filter(activity => {
+		const parent = activity.parent;
+
+		if (parentId) {
+			return parent && utils.uuidToString(parent) === parentId;
+		} else {
+			return !parent;
+		}
+	}).map(activity => {
+		const subActivities = getSubActivities(utils.uuidToString(activity._id), activities);
+		return  simplifyActivity(activity, subActivities);
+	});
+};
+
+const createActivitiesTree = async(account, model, sequenceId) => {
+	let activities = {};
+	const foundActivities = await db.find(account, model + ".activities",{sequenceId: utils.stringToUUID(sequenceId)}); // filter by sequenceId
```

Comment:
I have to check which actual fields are being use (for example parent is being use but get rid in simplify activiy). But some sort of simplify activity should be in order because it transforms the plain activity to a tree like activity (using parent and no subactivities).
Maybe I should change the name, because is does not reflect exactly what it does.

Reply:
but I think we def don't need the metadata part right? If that's not used then I think we should at least omit that as that's the bulky part

you can do projection with exclusion instead of inclusion, i.e. `{metadata: 0}`. That should return everything but the fields you specified.